libsql-server/src/hrana/ws/conn.rs

```diff
@@ -206,15 +206,13 @@ async fn handle_client_msg(conn: &mut Conn, client_msg: proto::ClientMsg) -> Res
 }
 
 async fn handle_hello_msg(conn: &mut Conn, jwt: Option<String>) -> Result<bool> {
+    let auth = session::handle_hello(&conn.server, jwt, conn.namespace.clone()).await;
+
     let hello_res = match conn.session.as_mut() {
-        None => {
-            session::handle_initial_hello(&conn.server, conn.version, jwt, conn.namespace.clone())
-                .await
-                .map(|session| conn.session = Some(session))
-        }
-        Some(session) => {
-            session::handle_repeated_hello(&conn.server, session, jwt, conn.namespace.clone()).await
-        }
+        None => auth
+            .map(|auth| session::Session::new(auth, conn.version))
+            .map(|s| conn.session = Some(s)),
+        Some(session) => auth.map(|a| session.update_auth(a)).and_then(|op| op),
```

Comment:
I'm wondering if the code wouldn't be a bit easier to follow if we structure it like this:
```
let hello_res = match auth {
    Ok(a) => match conn.session.as_mut() {
        None => {
            conn.session = Some(session::Session::new(a, conn.version));
            Ok(())
        }
        Some(session) => session.update_auth(a)
    }
    Err(e) => Err(e)
}
```